x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -159,14 +151,39 @@ class AgentPolicyService {
     soClient: SavedObjectsClientContract,
     esClient: ElasticsearchClient,
     newAgentPolicy: NewAgentPolicy,
-    searchParams: {
-      searchFields: string[];
-      search: string;
-    }
+    searchParams:
+      | { id: string }
+      | {
+          searchFields: string[];
+          search: string;
+        }
   ): Promise<{
     created: boolean;
     policy: AgentPolicy;
   }> {
+    // For preconfigured policies with a specified ID
+    if ('id' in searchParams) {
+      try {
+        const agentPolicy = await soClient.get<AgentPolicySOAttributes>(
+          AGENT_POLICY_SAVED_OBJECT_TYPE,
+          searchParams.id
+        );
+        return {
```

Comment:
According to Typescript it's either a SavedObject or it throws an error so I'm not sure what those checks are for.

Reply:
Yeah, that's how I read it, too. I'm starting to think this might be leftover from when we used to have another layer between here and ES. I believe some of those wrapper methods _would_ return `null`. Regardless, no need to change anything here.